qiskit/circuit/quantumcircuit.py

```diff
@@ -1533,6 +1534,381 @@ def _rebind_definition(self, instruction, parameter, value):
+
+    def mcx(self, control_qubits, target_qubit, ancilla_qubits=None, mode='noancilla'):
+        """Apply :class:`~qiskit.circuit.library.standard_gates.MCXGate`.
+
+        The multi-cX gate can be implemented using different techniques, which use different numbers
+        of ancilla qubits and have varying circuit depth. These modes are:
+        - 'no-ancilla': Requires 0 ancilla qubits.
+        - 'recursion': Requires 1 ancilla qubit if more than 4 controls are used, otherwise 0.
+        - 'v-chain': Requires 2 less ancillas than the number of control qubits.
+        - 'v-chain-dirty': Same as for the clean ancillas (but the circuit will be longer).
+        """
+        from .library.standard_gates.x import MCXGrayCode, MCXRecursive, MCXVChain
+        num_ctrl_qubits = len(control_qubits)
+
+        available_implementations = {
+            'noancilla': MCXGrayCode(num_ctrl_qubits),
+            'recursion': MCXRecursive(num_ctrl_qubits),
+            'v-chain': MCXVChain(num_ctrl_qubits, False),
+            'v-chain-dirty': MCXVChain(num_ctrl_qubits, dirty_ancillas=True),
+            # outdated, previous names
+            'advanced': MCXRecursive(num_ctrl_qubits),
+            'basic': MCXVChain(num_ctrl_qubits, dirty_ancillas=False),
+            'basic-dirty-ancilla': MCXVChain(num_ctrl_qubits, dirty_ancillas=True)
+        }
+
+        # check ancilla input
+        if ancilla_qubits:
+            _ = self.qbit_argument_conversion(ancilla_qubits)
+
+        try:
+            gate = available_implementations[mode]
+        except KeyError:
+            all_modes = list(available_implementations.keys())
+            raise ValueError('Unsupported mode ({}) selected, choose one of {}'.format(mode,
+                                                                                       all_modes))
+
+        if hasattr(gate, 'num_ancilla_qubits') and gate.num_ancilla_qubits > 0:
+            required = gate.num_ancilla_qubits
+            if ancilla_qubits is None:
+                raise AttributeError('No ancillas provided, but {} are needed!'.format(required))
+
+            # convert ancilla qubits to a list if they were passed as int or qubit
+            if not hasattr(ancilla_qubits, '__len__'):
+                ancilla_qubits = [ancilla_qubits]
+
+            if len(ancilla_qubits) < required:
+                actually = len(ancilla_qubits)
+                raise ValueError('At least {} ancillas required, but {} given.'.format(required,
+                                                                                       actually))
+            # size down if too many ancillas were provided
+            ancilla_qubits = ancilla_qubits[:required]
+        else:
+            ancilla_qubits = []
+
+        return self.append(gate, control_qubits[:] + [target_qubit] + ancilla_qubits[:], [])
+
+    def mct(self, control_qubits, target_qubit, ancilla_qubits=None, mode='noancilla'):
```

Comment:
what does this gate do and why do we want to have it?

Reply:
old code that is going through deprecation (i think? @Cryoris ?)

Reply:
That's what the multi-controlled X gate was called (t for Toffoli). I'd be happy do deprecate it, this is part of the aliasing issue you opened